wip-0002.md

```diff
@@ -0,0 +1,212 @@
+
+
+### Tally transactions
+
+**`rewarded_witnesses` and `slashed_witnesses` fields.** The `rewarded_witnesses` field in the tally transaction data structure (`TallyTransaction`) is replaced by a new field, `slashed_witnesses`:
+
+```protobuf
+message TallyTransaction {
+    Hash dr_pointer = 1;
+    bytes tally = 2;
+    repeated ValueTransferOutput outputs = 3;
+    repeated PublicKeyHash slashed_witnesses = 4;
+}
+```
+
+**Rewarded identities.** In general terms, identities found in `outputs` can be considered rewarded identities. The only exception is the identity that signed the data request ("the requestor"), which must not be considered rewarded solely based in its presence in `outputs`. The requestor is only considered rewarded if in addition to being present in the `outputs` field, it is not in the `slashed_witnesses` field.
+
+**Slashed identities.** Conversely, all identities found in `slashed_witnesses` can be considered slashed, with no exception whatsoever.
+
+**Only committers are listed.** For an identity to be present in `outputs` or `slashed_witnesses`, one commitment transaction for the data request must exist in the block chain prior to the tally transaction itself, such that the commitment is signed by the same identity, with the only exception of the requestor, which can be present in `outputs` for a different reason (see *"Change output"* below).
+
+**All committers must be listed.** All identities that produced commitment transactions for a data request that exist in the block chain prior to the tally transaction itself ("committers") must be present in either one of the `outputs` or `slashed_witnessess` fields.
+
+**No identity repetition.** No identity can appear more than once in either the `output` or `slashed_identities` fields. In the same manner, no identity can be present in both the `outputs` and `slashed_identities` fields, with the only exception of the requestor, which in addition to being in one of those fields, can calso be present in (a) both at the same time OR (b) twice in `outputs` (see *"Change output"* below).
+
+**Requests with insufficient commits**. Given a data request, may not exist enough commitments for it in the `extra_commit_rounds + 1` number of blocks (as specified by the request) following the block in which it was bound to enter the commit stage, a tally can be immediately produced such that the result is the "Insufficient commits" RADON error, the `slashed_identities` field is empty, and the `outputs` field contains only the change output (see *"Change output"* below).
+
+**Missing reveals**. All committers for which no valid reveal transaction exists in the block chain prior to the tally transaction itself must be listed in the`slashed_witnessess` field. To this effect, for a reveal transaction to be considered valid, it must be properly constructed such that it can be decoded into a compliant RADON type or RADON error.
```

Comment:
In case of none reveals, collateral comes back to the committers but they will be slashed only in reputation

Reply:
Great point. It is now addressed by b415858736e277c58876d10e61f6351e6526cea2.